server/app/services/program/ProgramServiceImpl.java

```diff
@@ -210,7 +216,17 @@ public ErrorAnd<ProgramDefinition, CiviFormError> updateLocalization(
   }
 
   private boolean hasProgramNameCollision(String programName) {
-    return getActiveAndDraftPrograms().getProgramNames().contains(programName);
+    ImmutableSet<String> programNames = getActiveAndDraftPrograms().getProgramNames();
+
+    if (programNames.contains(programName)) {
+      return true;
+    }
+
+    Slugify slugifier = new Slugify();
+    ImmutableSet<String> programSlugs =
+        programNames.stream().map(slugifier::slugify).collect(ImmutableSet.toImmutableSet());
+
+    return programSlugs.contains(slugifier.slugify(programName));
```

Comment:
I think we can reduce/simplify this method to only this check.  Any two program names that collide above will collide here too.

Reply:
something like

```
String newProgramSlug = slugifier.slugify(programName);

return getActiveAndDraftPrograms()
.getProgramNames()
.stream
.map(slugifier::slugify)
.anyMatch(newProgramSlug::equals);
```

Reply:
done